src/software/jetson_nano/display/screens/chip_and_kick_screen.py

```diff
@@ -0,0 +1,142 @@
+from screen import Screen
+
+CHIP = 0
+KICK = 1
+
+PADDING = 6
+BASE_Y = 20
+
+"""
+This screen is used to edit the chip and kick speed settings
+"""
+
+
+class ChipAndKickScreen(Screen):
```

Comment:
WheelsScreen and ChipAndKickScreen share a lot of code in common

Do you think its possible to consolidate the functionality into `Screen`

The constructor would take in the settings like:
~~~python
[              
         {
                    "param": "chip speed"
                    "type": double,
                    "delta": 0.5,
                    "redis key": "chip speed",
           },
           {
                    "param": "kick speed",
                    "type": double,
                    "delta": 0.5,
                    "redis key": "kick speed",
             }
]
~~~

All settings will either be bools or doubles, so you could pass in the type name

You could still have this class to do additional functionality based on the state of these settings

We could even make the param name and the redis key the same to make it more concise

Reply:
That way adding new screens to the `Menu` page would just be creating instances of this consolidated class

Reply:
Each screen has their own `draw_screen()` function which is pretty custom for each screen so I think it is best to keep the parent and child class relationship.  Additionally, the `Menu` file would become quite long if we had all the screen instances being defined in the same file I think.

However, I have refactored the screen files so that there is no need for all the callback functions and the actions are now in a format that is similar to the one you described. This makes the code more concise and simpler to implement new screens in the future. I think a drawback of this change would be that the callback functions allow more customizability of the actions that could be performed but this wouldn't be necessary if the UI will be used for just simple actions.